src/sema/contracts.rs

```diff
@@ -15,11 +15,18 @@ use crate::emit;
 
 impl ast::Contract {
     /// Create a new contract, abstract contract, interface or library
-    pub fn new(name: &str, ty: pt::ContractTy, tags: Vec<ast::Tag>, loc: pt::Loc) -> Self {
+    pub fn new(
+        name: &str,
+        ty: pt::ContractTy,
+        tags: Vec<ast::Tag>,
+        loc: pt::Loc,
+        math_overflow_check: bool,
```

Comment:
I don't think this new parameter is necessary. The optimization is also not specified, which is also a codegen issue (and not sema).